RFCs/0016-Loggers.md

```diff
@@ -0,0 +1,85 @@
+# 0016 Loggers
+
+## Summary
+This note outlines the proposed changes for:
+1. Enabling logger support in test platform for TranslationLayer clients.
+2. Allowing users to provide loggers from the runsettings.
+
+## Motivation
+1. Enabling logger support for TranslationLayer clients will allow clients to use logger extensibility feature.
+
+2. Allowing users to provide loggers from the runsettings enables:
+  * TranslationLayer clients to provide loggers to the test platform.
+  * Alternative way to provide loggers in addition to the existing command line logger argument.
+
+## Specifying a logger
+Loggers can be provided to the test platform in one of the following ways:
+
+1.  "/Logger:"UriOrFriendlyName";Key1=Value1;Key2=Value2" This is a switch to vstest.console.exe that feeds logger information to the test platform. For instance:
+
+    ```
+    /Logger:"logger://Microsoft/TestPlatform/TrxLogger/v1"
+    ```
+
+    This loads and initializes logger with "logger://Microsoft/TestPlatform/TrxLogger/v1" as uri.
+
+    ```
+    /Logger:sampleLogger;Key1=Value1;Key1=Value2
+    ```
+
+    This loads and initializes logger with "sampleLogger" as friendly name. Key1=Value1 and Key2=Value2 are passed as dictionary parameters to the logger while initialization.
+
+2. Runsettings via "Logger" node in the LoggerRunSettings section. Here is a sample on how this can be specified:
+
+    ```xml
+    <RunSettings>
+        <LoggerRunSettings>
+            <Loggers>
+                <Logger friendlyName="sampleLoggerwithParameters">
+                    <Configuration>
+                        <Key1>Value1</Key1>
+                        <Key2>Value2</Key2>
+                    </Configuration>
```

Comment:
Did we consider passing the entire runsettings to each logger? This will allow a logger to have any custom parameter as required (similar to Datacollectors).

Reply:
This requires API changes in logger extensibility. We are not taking this change as part of this RFC. But we are taking key values in configuration element from runsettings thinking these changes in mind.

Reply:
Makes sense.